enterprise/admin/__init__.py

```diff
@@ -831,22 +833,59 @@ def get_fields(self, request, obj=None):
         return fields
 
 
+class BigTableMysqlPaginator(Paginator):
+    """
+    A paginator that uses INFORMATION_SCHEMA.TABLES to estimate
+    the total number of rows in a table.
+    """
+    ARBITRARILY_LARGE_NUMBER = 10000000
+
+    # pylint: disable=attribute-defined-outside-init
+    @property
+    def count(self):  # pylint: disable=invalid-overridden-method
+        """
+        Returns the number of items in the object list (possibly an estimate).
+        """
+        query = self.object_list.query
+
+        if query.where:
```

Comment:
[clarification] Does `query.where` also include non-search filters (e.g., by role)?

![image](https://user-images.githubusercontent.com/2828721/115048584-b8145780-9ea7-11eb-98b5-28e4f9bb92a1.png)

Reply:
Yeah, it does, and the example you provided will be a pretty large queryset in prod (assuming there's no search term to go along with that filter).  I'm inclined to say "let's see what happens" for now.